### PR TITLE
Mepts |1055 Changes MQ Cat 15

### PR DESCRIPTION
**Pull request by eliazmuchave**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 1]
```diff
@@ -8441,4 +8827,539 @@ public CohortDefinition getMQC13DEN_B2E() {
...
+            + "               AND e.encounter_datetime <= :endDate "
+            + "               AND e.encounter_datetime >= (SELECT dt_date "
+            + "                                            FROM   (SELECT  most_recent.patient_id, MAX(most_recent.encounter_datetime) dt_date "
+            + "                                                    FROM   (SELECT p2.patient_id, e2.encounter_datetime "
+            + "                                                             FROM   patient p2 "
+            + "                                                             INNER JOIN encounter e2 ON e2.patient_id = p2.patient_id "
+            + "                                                             INNER JOIN obs otype ON otype.encounter_id = e2.encounter_id "
+            + "                                                             INNER JOIN obs ostate ON ostate.encounter_id = e2.encounter_id "
+            + "                                                    WHERE  e2.encounter_type = 6 "
+            + "                                                      AND e2.location_id = 5 "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: @eliazmuchave I think you forgot this!

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 2]
```diff
@@ -8441,4 +8827,539 @@ public CohortDefinition getMQC13DEN_B2E() {
...
+
+  /**
+   * Filtrando os utentes que têm o registo de dois pedidos de CV na Ficha Clínica (“Pedido de
+   * Investigações Laboratoriais” igual a “Carga Viral”) entre “Data Inscrição MDS (DT)” e “Data Fim
+   * Revisão” e o registo de um resultado de CV <1000 cps/ml entre os dois pedidos (“Data Consulta
+   * Resultado CV”> “Data Consulta 1º Pedido de CV” e < “Data Consulta 2º Pedido de CV”)
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getPatientsWithVLResultLessThan1000Between2VlRequestAfterTheseMDS(
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:9137]: Let's discuss in the morning @eliazmuchave

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@eliazmuchave we need to look the queries together

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java [hunk 3]
```diff
@@ -1838,4 +1920,49 @@ public static String getRestartedArtQuery() {
...
+        "   SELECT p.patient_id "
+            + "  FROM patient p  "
+            + "  INNER JOIN encounter e ON p.patient_id = e.patient_id  "
+            + "  INNER JOIN obs o ON e.encounter_id = o.encounter_id  "
+            + "  WHERE  p.voided = 0   "
+            + "         AND e.voided = 0   "
+            + "         AND o.voided = 0     "
+            + "         AND o.concept_id = ${856}   "
+            + "         AND o.value_numeric >= 1000  "
+            + "         AND ( e.encounter_type = ${6} AND e.encounter_datetime BETWEEN :startDate AND :endDate) "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]: No parentesis

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java [hunk 4]
```diff
@@ -1838,4 +1920,49 @@ public static String getRestartedArtQuery() {
...
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(map);
+
+    sqlCohortDefinition.setQuery(stringSubstitutor.replace(query));
+
+    return sqlCohortDefinition;
+  }
+
+  private static String getMetadataFrom(List<Integer> dispensationTypes) {
+    if (dispensationTypes == null || dispensationTypes.isEmpty()) {
+      throw new RuntimeException("The list of encounters or concpts might not be empty ");
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]:
```suggestion
      throw new RuntimeException("The list of encounters or concepts might not be empty ");
```

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java (hunk 3 above)
Comment (eliazmuchave) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java]: Fixed

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**